### DISCUSSION TITLE
api timesheets problem

### DISCUSSION THREAD

**Discussion by UBT-FJ** [kimai/kimai#3528] [Questions & Answers]
Hello,

I'm trying to create a record in timesheets through api but always get this

`{
    "code": 400,
    "message": "Validation Failed",
    "errors": {
        "children": {
            "project": {
                "errors": [
                    "A project needs to be selected.",
                    "This value should not be null."
                ]
            },
            "activity": {
                "errors": [
                    "An activity needs to be selected.",
                    "This value should not be null."
                ]
            },
            "description": {},
            "tags": {}
        }
    }
}`

According to the docs in api the required fields are these:

`{
        "activity": 2,
        "project": 2,
        "begin": "2022-09-09T21:21:00+0200"
    }`

But always get that error, I can do GET but not POST method!

I also tried these fields, but I get the same result!

`{
  "begin": "2022-09-09T21:22:00",
  "end": "2022-09-09T21:22:00",
  "project": 2,
  "activity": 2,
  "description": "string",
  "fixedRate": 0,
  "hourlyRate": 0,
  "user": 3,
  "exported": true,
  "billable": true,
  "tags": "string"
}`

P.S. the project, activity and user exist in the system and I have enabled the api for that user.

**Comment by kevinpapst** [comment]
Please show the exact request you are posting, e.g. create it with POSTMAN and export it as curl.

**Reply by UBT-FJ** [reply]
Hello @kevinpapst , thanks for the quick reply.

`curl -X POST url/api/timesheets \
    -H 'X-AUTH-USER: user' \
    -H 'X-AUTH-TOKEN: token' \
    -H 'Accept: application/json' \
    -d '{
        "activity": 2,
        "project": 2,
        "begin": "2022-09-09T21:21:00+0200"
    }'`

I don't use postman, I use rester in chrome.

**Reply by kevinpapst** [reply]
Looks correct. Try to use curl or another tool (like postman).
If the error persists, check your webserver configuration.
You can always use the demo installation to verify the issue: https://www.kimai.org/demo/

**Reply by UBT-FJ** [reply]
I have tried from the demo and it is working, but not in my domain! 

One thing I'm missing, would this not work without https? because I only have http.

**Reply by UBT-FJ** [reply]
I tried now directly from curl but get the same error
`curl -X POST https://demo-plugins.kimai.org/api/timesheets \
    -H 'X-AUTH-USER: admin' \
    -H 'X-AUTH-TOKEN: na8GtB9Q3Phh2IZuiTPkwFdHhkDSN6qz' \
    -H 'Accept: application/json' \
    -d '{
  "begin": "2022-09-10T16:56:09",
  "project": 84,
  "activity": 1085,
  "description": "string",
  "fixedRate": 0,
  "hourlyRate": 0,
  "user": 31,
  "exported": true,
  "billable": true,
  "tags": "string"
}'
{"code":400,"message":"Validation Failed","errors":{"errors":["This form should not contain extra fields."],"children":{"begin":{},"end":{},"project":{"errors":["A project needs to be selected.","This value should not be null."]},"activity":{"errors":["An activity needs to be selected.","This value should not be null."]},"description":{},"fixedRate":{},"hourlyRate":{},"user":{},"exported":{},"billable":{},"tags":{}}}}% `

But from /api/doc it works, but with my domain it doesnt work, I always get this message, and this is just for the post method.

**Comment by UBT-FJ** [comment]
Hello, I just did a new installation from scratch and still there is the same error!!

**Reply by kevinpapst** [reply]
Sorry, but there is nothing I can do. 
It is not a bug in Kimai, but a problem with your environment/setup/config/docker/whatever.

**Reply by UBT-FJ** [reply]
Hello @kevinpapst , thanks for the quick reply, the installation is straight forward all according to the docs! really strange, this is keeping me from using the software as I want to send api calls from the rfid microcontroller.

**Reply by kevinpapst** [reply]
The demo installation is the same. Just a straight forward automatic installation from master branch.
If you want help, you should post more infos about your environment. Posting "it doesn't work" includes no information whatsoever. I still don't know whether you use Docker or nginx with fpm or Apache mit mod-php
